sdk/src/main/java/io/radar/sdk/RadarLocationManager.kt

```diff
@@ -576,16 +575,21 @@ internal class RadarLocationManager(
     private fun startForegroundService(foregroundService: RadarTrackingOptions.RadarTrackingOptionsForegroundService) {
         if (Build.VERSION.SDK_INT >= 26) {
             try {
-                val intent = Intent(context, RadarForegroundService::class.java)
-                intent.action = "start"
-                intent.putExtra("id", foregroundService.id)
+                if (RadarForegroundService.started) {
```

Comment:
Can you help me understand why moving the `started` logic into this function helps?

Reply:
Previously we called `Context.startForegroundService()`,  which called `onStartCommand()` of `RadarForegroundService`. We then checked the `started` flag inside `onStartCommand()` to determine whether to call `Service.startForeground()`.

However, if you call `Context.startForegroundService()` and don't call `Service.startForeground()` within 5 seconds, Android throws a `RemoteServiceException`.

With this change, we check the `started` flag before calling `Context.startForegroundService()`, which prevents the `RemoteServiceException`.

Reply:
This should be a big improvement.  One thing I'm thinking about is multiple callers calling this function and the race condition between when `RadarForegroundService.started` is read and when it's set to `true`.  In a future version we may want to use an `AtomicBoolean.compareAndSet()`, or similar approach, depending on the impact of starting multiple foreground services and how often this race happens.